### PR TITLE
[Restoration Shaman] Redo Unleash Life

### PR DESCRIPTION
**Pull request by vohrr**
### Description

Rewrote Unleash Life using event linking for the underlying interactions between the buff and the affected spells

### Motivation

The current iteration of this module was inaccurate and only partially implemented. It is now much more accurate and analyzes all spells that the talent affects
### Testing
`/report/H9VDyfZ4FKABNtGk/36-Mythic+Kurog+Grimtotem+-+Kill+(8:49)/Fakeblonde/standard/statistics` - most common
`/report/PdXj1MctQk63fAWZ/16-Mythic+Terros+-+Kill+(5:34)/Dalfaëran/standard/statistics`  - all but healing surge
`report/c8BjmPGdf2gHQAVM/27-Mythic+Terros+-+Kill+(6:41)/17-Bigdaddyjeff/standard/statistics` - healing surge

### Screenshot(s):
Before
image

After
image

### CODE REVIEW

**Comment by vohrr** [comment]
Added spellbreakdowns in tooltips for each: 
image

File: src/analysis/retail/shaman/restoration/modules/talents/UnleashLife.tsx [hunk 1]
```diff
@@ -43,254 +75,512 @@ interface HealingBuff {
...
-      healing: 0,
-      castAmount: 0,
+      amount: 0,
+      casts: 0,
     },
     [TALENTS.DOWNPOUR_TALENT.id]: {
-      healing: 0,
-      castAmount: 0,
+      amount: 0,
+      casts: 0,
```
Comment (trevorm4) [inline src/analysis/retail/shaman/restoration/modules/talents/UnleashLife.tsx:113]: pretty messy, probably would be better to have a constant of UL spell ids and use it in multiple places.

**Review by trevorm4 — APPROVED** [review]
Overall lgtm, might want to consider sorting the list of spells for the chart for better readability, but otherwise looks good

File: src/analysis/retail/shaman/restoration/modules/talents/UnleashLife.tsx (hunk 1 above)
Comment (vohrr) [reply src/analysis/retail/shaman/restoration/modules/talents/UnleashLife.tsx:113]: Yeah this object was the one holdover from the old code and is used in healing efficiency and others so I refactored a bit but kept it in place to keep my own headache for this module lower

**Pull request merged**